+ set -o errtrace
+ python /rds/general/user/ga402/home/009_GAN/001_2D_DBGAN/dbgan2d.py --project /rds/general/user/ga402/home/PROJECTS/001_AMLCD8TC/DBGAN2d --dataroot /rds/general/user/ga402/home/PROJECTS/001_AMLCD8TC/IMAGES2D --workers 2 --batch_size 128 --image_size 128 --nc 3 --nz 100 --ngf 64 --ndf 64 --num_epochs 20 --lr 0.00001 --beta1 0.5 --ngpu 1 --accessory_info 'DBGAN2d; Wasserstein GAN attempt'
[W Context.cpp:70] Warning: torch.use_deterministic_algorithms is in beta, and its design and functionality may change in the future. (function operator())
Traceback (most recent call last):
  File "/rds/general/user/ga402/home/009_GAN/001_2D_DBGAN/dbgan2d.py", line 269, in <module>
    save_fake_images,
  File "/rds/general/user/ga402/home/009_GAN/001_2D_DBGAN/utils/wgan_utils.py", line 137, in w_fit
    gen, disc, disc_optimiser, images, batch_size, z_dim, device
  File "/rds/general/user/ga402/home/009_GAN/001_2D_DBGAN/utils/wgan_utils.py", line 81, in w_train_discriminator
    disc_loss.backward(retain_graph=True)
  File "/rds/general/user/ga402/home/miniconda/envs/yoloV5_model/lib/python3.6/site-packages/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/rds/general/user/ga402/home/miniconda/envs/yoloV5_model/lib/python3.6/site-packages/torch/autograd/__init__.py", line 147, in backward
    allow_unreachable=True, accumulate_grad=True)  # allow_unreachable flag
RuntimeError: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithms(true)`, but this operation is not deterministic because it uses CuBLAS and you have CUDA >= 10.2. To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
